platform/view/services/db/driver/badger/badger.go

```diff
@@ -7,104 +7,61 @@ SPDX-License-Identifier: Apache-2.0
 package badger
 
 import (
-	"bytes"
-	"strings"
 	"sync"
+	"time"
 
 	"github.com/dgraph-io/badger/v3"
 	"github.com/golang/protobuf/proto"
-	"github.com/pkg/errors"
-
-	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/cache/secondcache"
-	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver"
 	dbproto "github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/badger/proto"
 	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/keys"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/pkg/errors"
 )
 
-var (
-	cacheEmptyProtoValue = &dbproto.VersionedValue{}
-)
-
-type cacheValue struct {
-	v *driver.VersionedRead
-	k []byte
-}
-
-type cache interface {
-	Get(key string) (interface{}, bool)
-	Add(key string, value interface{})
-	Delete(key string)
-}
-
-type ItemList struct {
-	items []cacheValue
-	sync.RWMutex
-}
-
-func (i *ItemList) Get(index int) (*driver.VersionedRead, bool) {
-	i.RLock()
-	defer i.RUnlock()
-	if index < 0 || index >= len(i.items) {
-		return nil, false
-	}
-	return i.items[index].v, true
-}
-
-func (i *ItemList) Set(index int, v *driver.VersionedRead, k []byte) {
-	i.Lock()
-	defer i.Unlock()
-
-	// if not in capacity, then skip
-	if index < 0 || index+1 > cap(i.items) {
-		return
-	}
-
-	// if not in length, then append
-	if index >= len(i.items) {
-		i.items = i.items[:index+1]
-	}
-
-	i.items[index].v = v
-	i.items[index].k = k
-}
-
-func (i *ItemList) GetLast() []byte {
-	i.RLock()
-	defer i.RUnlock()
-	if len(i.items) == 0 {
-		return nil
-	}
-	return i.items[len(i.items)-1].k
-}
+var logger = flogging.MustGetLogger("db.driver.badger")
 
 type badgerDB struct {
-	db *badger.DB
-
+	db      *badger.DB
 	txn     *badger.Txn
-	deletes []string
 	txnLock sync.RWMutex
-	cache   cache
-
-	itemsListSize int
-	itemsMap      map[string]*ItemList
 }
 
 func OpenDB(path string) (*badgerDB, error) {
 	if len(path) == 0 {
 		return nil, errors.Errorf("path cannot be empty")
 	}
 
-	db, err := badger.Open(badger.DefaultOptions(path))
+	// let's pass our logger badger
+	opt := badger.DefaultOptions(path)
+	opt.Logger = logger
+
+	db, err := badger.Open(opt)
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not open DB at '%s'", path)
 	}
 
-	return &badgerDB{
-		db:            db,
-		cache:         secondcache.New(20000),
-		itemsMap:      map[string]*ItemList{},
-		itemsListSize: 20000,
-	}, nil
+	// start our auto cleaner
+	autoCleaner(db)
+
+	return &badgerDB{db: db}, nil
+}
+
+func autoCleaner(db *badger.DB) {
+	go func() {
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for range ticker.C {
+		again:
+			if db.IsClosed() {
+				// no need to clean anymore
+				return
+			}
+			err := db.RunValueLogGC(0.7)
+			if err == nil {
```

Comment:
does this mean that if no error happens, RunValueLogGC is called continuously? 
Should `again` come before the for?